torch_xla/test/metrics_test_wrapper.py

```diff
@@ -112,21 +150,69 @@ def _write_to_disk(output_string, output_filename):
   if not FLAGS.root or not FLAGS.test_folder_name:
     raise ValueError('root and test_folder_name are required arguments.')
 
-  # TODO(zcain): Verify that root contains base_config.
-  # TODO(zcain): Read tolerances from base_config and maybe override with
-  #              test-specific config from test_folder_name.
-  # TODO(zcain): Read metrics_history for current test, retrieve N days of
-  #              history.
-  # TODO(zcain): Calculate mean and std dev for each metrics, then compare
-  #              against metrics from the subprocess call.
-  
+  # Run the user-supplied command.
   metrics, sp_return_code = _run_subprocess(FLAGS.positional)
 
-  # Include the params for this invocation when saving metrics.
-  output_string = '{}\n\n{}'.format(FLAGS, metrics)
-  output_filename = os.path.join(
-      FLAGS.root, FLAGS.test_folder_name, _METRICS_HISTORY_DIR_NAME,
-      datetime.datetime.utcnow().strftime('%Y_%m_%d'))
-  _write_to_disk(output_string, output_filename)
-
+  # Retrieve any config files that affect this test.
+  # NOTE: these are ordered in increasing specificity. For example, if there
+  # was a base config that affects all tests and a specific config for a
+  # particular test, then the base config will be the first element in the
+  # list and the most specific config will be the last element.
+  ordered_config_dicts = []
+
+  path_to_search = FLAGS.test_folder_name
+  while True:
+    try:
+      f = gcsfs.open(os.path.join(FLAGS.root, path_to_search, 'config.json'))
+      ordered_config_dicts.insert(0, json.load(f))
+    except google.api_core.exceptions.NotFound:
+      pass
+    if path_to_search == '':
+      break
+    path_to_search = os.path.split(path_to_search)[0]
+  if not ordered_config_dicts:
+    print('No config files found. See example usage at top of '
+          'metrics_test_wrapper.py')
+    sys.exit(1)  # Return non-OK status since config is required.
```

Comment:
Module functions should not exit().

Reply:
What is the preferred way to indicate a non-0 exit code so that the caller of this script knows something is wrong?

Reply:
Module functions should `raise` on errors.

Reply:
You `raise` with proper error message, not printing it.
If you want to have different exit codes depending on exception types, you have an try:catch: in main().

Reply:
I will switch to `raise`, but for my understanding: is this considered a module function if it occurs inside `if __name__ == '__main__':` ?